CMakeLists.txt

```diff
@@ -519,18 +535,7 @@ if (BUILD_TESTING)
         endif()
         add_test(NAME ${test_case_name} COMMAND $<TARGET_FILE:${test_case_name}> WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests/unit)
         set_property(TEST ${test_case_name} PROPERTY LABELS "unit")
-
-        set_property(
-        TEST
-            ${test_case_name}
-        PROPERTY
-            ENVIRONMENT LD_PRELOAD=$<TARGET_FILE:allocator_overrides>)
-
-        set_property(
-        TEST
-            ${test_case_name}
-        PROPERTY
-            ENVIRONMENT S2N_DONT_MLOCK=1)
+        set_property(TEST ${test_case_name} PROPERTY ENVIRONMENT ${UNIT_TEST_ENVS})
```

Comment:
Setting [ENVIRONMENT](https://cmake.org/cmake/help/latest/prop_test/ENVIRONMENT.html) like this overrides instead of appends. So the first call to set LD_PRELOAD has no affect on how the tests are ultimately run, but the call to S2N_DONT_MLOCK does. CMake 3.22 added [ENVIRONMENT_MODIFICATION](https://cmake.org/cmake/help/latest/prop_test/ENVIRONMENT_MODIFICATION.html) to support modifying instead of completely overriding, but our CI uses some very old versions of CMake and we shouldn't assume our customers are using the latest either.

I fixed this problem by constructing the full set of environment variables as a string so that I could set ENVIRONMENT once.